terraform/modules/opensearch_domain/opensearch.tf

```diff
@@ -0,0 +1,13 @@
+resource "aws_opensearch_domain" "opensearch" {
```

Comment:
We need to add support for a few more things here:

- advanced security options: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/opensearch_domain#advanced_security_options. we should allow a variable for `internal_user_database_enabled` but default it to true. we should allow specifying a `master_user_name` and `master_user_password` via variables.

We should also add:

```
  node_to_node_encryption {
    enabled = true
  }
  
  encrypt_at_rest {
    enabled = true
  }  
```

Reply:
sure!!